lxd/container.go

```diff
@@ -587,6 +588,64 @@ func containerCreateAsEmpty(d *Daemon, args db.ContainerArgs) (container, error)
 	return c, nil
 }
 
+func containerCreateFromBackup(s *state.State, info backupInfo, data []byte) error {
+	var pool storage
+	var fixBackupFile = false
+
+	// Get storage pool from index.yaml
+	pool, storageErr := storagePoolInit(s, info.Pool)
+	if storageErr != nil && storageErr != db.ErrNoSuchObject {
+		// Unexpected error
+		return storageErr
+	}
+
+	if storageErr == db.ErrNoSuchObject {
+		// The pool doesn't exist, and the backup is in binary format so we
+		// cannot alter the backup.yaml.
+		if info.HasBinaryFormat {
+			return storageErr
+		}
+
+		// Get the default profile
+		_, profile, err := s.Cluster.ProfileGet("default")
+		if err != nil {
+			return err
+		}
+
+		_, v, err := shared.GetRootDiskDevice(profile.Devices)
+		if err != nil {
+			return err
+		}
+
+		// Use the default-profile's root pool
+		pool, err = storagePoolInit(s, v["pool"])
+		if err != nil {
+			return err
+		}
+
+		fixBackupFile = true
+	}
+
+	// Unpack tarball
+	err := pool.ContainerBackupLoad(info, data)
+	if err != nil {
+		return err
+	}
+
+	if fixBackupFile {
+		// Use the default pool since the pool provided in the backup.yaml
+		// doesn't exist.
+		err = fixBackupStoragePool(s.Cluster, info)
+		if err != nil {
+			return err
+		}
+	} else {
+		return err
+	}
+
+	return nil
```

Comment:
I'm confused, couldn't you just remove the `else`  and `return nil` or are you intending to return an actual error? If the latter this won't work right now since reaching this branch means that the previous function call that altered `err` succeded.

Reply:
Yeah, that doesn't really make sense, the else part can just be removed, I'm surprised none of the static analysis tools picked that up

Reply:
fixed